Directory.Build.props

```diff
@@ -247,6 +247,7 @@
     <!-- Disable source link when building locally. -->
     <DisableSourceLink Condition="'$(DisableSourceLink)' == '' and
                                   '$(ContinuousIntegrationBuild)' != 'true' and
+                                  '$(DotNetBuildFromSource)' != 'true' and
```

Comment:
Would it be better to explicitly specify this property in the [source-build configuration](https://github.com/dotnet/runtime/blob/main/eng/SourceBuild.props#L34) vs a sourcebuild condition here?

Reply:
The benefit is it utilizes the existing mechanism for source-build and reduces the source-build specific checks throughout the code.

Reply:
> Would it be better to explicitly specify this property in the [source-build configuration](https://github.com/dotnet/runtime/blob/main/eng/SourceBuild.props#L34) vs a sourcebuild condition here?

Makes sense - moving it there.